src/types.rs

```diff
@@ -348,6 +350,17 @@ pub struct OrderState {
 pub struct NewOrder<'a> {
     /// Identifiers to be included in the order
     pub identifiers: &'a [Identifier],
+    /// An optional `EncodedCertificateIdentifier` that this order is replacing
+    ///
+    /// Some ACME servers may give preferential rate limits to orders that replace
+    /// existing certificates, or use this information to determine when it is safe
+    /// to revoke a certificate affected by a compliance incident.
+    ///
+    /// When provided, at least one of the `identifiers` for the new order must have been
+    /// present in the certificate being replaced. If the ACME CA does not support the
+    /// ACME renewal information (ARI) extension, the [Account::new_order()] method will
+    /// return an error.
+    pub replaces: Option<EncodedCertificateIdentifier>,
```

Comment:
How do you feel about inserting a commit before this that changes it from having `pub` fields to having a `new(identifiers: &'a [Identifier])`, and then inserting this as a builder method `.replaces()`? Many/most existing callers will not care to specify `replaces: None`.

Reply:
Good idea. Done.

I also added some commits later on to switch the Pebble common test logic to accept a `&NewOrder` instead of `&[&'static str]` identifiers. This let those tests avoid unnecessary `None` arguments in a similar manner.